### PR TITLE
some python version compatibility changes

### PR DESCRIPTION
**Pull request by outofculture**
Fixed some iteritems, buffer, raise, basestring. Oh! And another docker to build a python3 environment.

### CODE REVIEW

File: acq4/devices/PatchPipette/statemanager.py [hunk 1]
```diff
@@ -139,7 +139,7 @@ def configureState(self, state, *args, **kwds):
                 self.configureState(oldJob.stateName, _allowReset=False)
             except Exception:
                 printExc("Error occurred while trying to reset state from a previous error:")
-            raise exc[0], exc[1], exc[2]
+            raise exc[0](exc[1]).with_traceback(exc[2])
```
Comment (campagnola) [inline acq4/devices/PatchPipette/statemanager.py]: I'm having a hard time remembering why I didn't just `raise exc[1]` here..

Comment (campagnola) [reply acq4/devices/PatchPipette/statemanager.py]: In any case, I don't think `with_traceback` is backward compatible.

Comment (campagnola) [reply acq4/devices/PatchPipette/statemanager.py]: maybe `six.reraise` ?

Comment (outofculture) [reply acq4/devices/PatchPipette/statemanager.py]: sixed.

**Pull request merged**